pulsar-client-cpp/include/pulsar/ConsumerConfiguration.h

```diff
@@ -33,6 +33,7 @@ class PulsarWrapper;
 
 /// Callback definition for non-data operation
 typedef boost::function<void(Result result)> ResultCallback;
+typedef boost::function<void(const Message& msg)> ReceiveCallback;
```

Comment:
What if consumer is not initialized - most of the other async operations in Consumer.cc return a ResultConsumerNotInitialized .